### PR TITLE
Check for debugging tools

### PR DESCRIPTION
**Pull request by TomMizrachi**
Test if developers forgot binding.pry in their code, and alert them accordingly

### CODE REVIEW

File: src/bindingPry/spec.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+} = require('.');
+
+const DIFF_CONTAINS_BINDING_PRY= `
+    blabla
+    +  binding.pry
+    blabla
+    blabla
+`;
+
+describe('dangerousSetInnerHTMLWarn', () => {
```
Comment (raphaelboukara) [inline src/bindingPry/spec.js]: add here the name of your function

File: src/bindingPry/spec.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+            test('should not call warn', () =>
+                run(files, diffForFile, warn)
+                    .then(() => {
+                        expect(warn).not.toHaveBeenCalled();
+                    })
+            );
+        });
+
+        describe('when a file contain "binding.pry"', () => {
```
Comment (raphaelboukara) [inline src/bindingPry/spec.js]: `contain` -> `contains`

File: src/dangerousSetInnerHTMLWarn/index.js [hunk 3]
```diff
@@ -20,10 +20,10 @@ const dangerouslySetInnerHTMLDetected = (str) =>
  * Warn if detect "dangerouslySetInnerHTML" added.
  * @param {String[]} files - list of modified files.
  * @param {Function} diffForFile - danger diffForFile.
- * @param {Function} warn - danger warn.
+ * @param {Function} fail - danger fail.
```
Comment (raphaelboukara) [inline src/dangerousSetInnerHTMLWarn/index.js:23]: thanks ❤️

File: dangerfile.js [hunk 4]
```diff
@@ -18,13 +18,15 @@ const {
     }
 } = danger;
 
+const bindingPry = require('./src/bindingPry');
```
Comment (raphaelboukara) [inline dangerfile.js]: rename to `bindingPryPresenceWarn` please

**Review by raphaelboukara — CHANGES_REQUESTED** [review]
Very nice, some style changes only

**Comment by omrilotan** [comment]
@TomMizrachi I just want to say I think this is a horrible contribution

File: dangerfile.js [hunk 5]
```diff
@@ -18,13 +18,15 @@ const {
     }
 } = danger;
 
+const bindingPryPresenceFail = require('./src/bindingPryPresenceFail');
```
Comment (raphaelboukara) [inline dangerfile.js]: bindingPryPresenceWarn!!! not FAIL

File: src/dangerousSetInnerHTMLWarn/index.js (hunk 3 above)
Comment (raphaelboukara) [inline src/dangerousSetInnerHTMLWarn/index.js:23]: it's a warn!! not a fail! 😂, warn was good

**Review by raphaelboukara — CHANGES_REQUESTED** [review]

File: src/dangerousSetInnerHTMLWarn/index.js (hunk 3 above)
Comment (omrilotan) [reply src/dangerousSetInnerHTMLWarn/index.js:23]: I disagree

Comment (omrilotan) [reply src/dangerousSetInnerHTMLWarn/index.js:23]: A specific route which got a `binding.pry` by mistake. This specific route had no response time alert and was discovered by chance

Comment (omrilotan) [reply src/dangerousSetInnerHTMLWarn/index.js:23]: The last drop is **removing binding.pry** only

File: src/bindingPryPresenceFail/index.js [hunk 6]
```diff
@@ -5,7 +5,7 @@
  * @default
  */
 const MESSAGE = `<b>Performance!</b> - <i>
-"binding.pry" may cause performance hits on production environments.
+"binding.pry" will cause performance hits on production environments.
```
Comment (omrilotan) [inline src/bindingPryPresenceFail/index.js:8]: @michael5r Cool. Much better

File: src/bindingPryPresenceFail/index.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ * @returns {Boolean}
+ */
+const bindingPryDetected = (str) =>
+    str && str.includes('binding.pry');
+
+/**
+ * Warn if detect "binding.pry" added.
+ * @param {String[]} files - list of modified files.
+ * @param {Function} diffForFile - danger diffForFile.
+ * @param {Function} warn - danger warn.
```
Comment (raphaelboukara) [inline src/bindingPryPresenceFail/index.js]: fail

File: src/bindingPryPresenceFail/index.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+ * @returns {Promise<undefined>}
+ */
+const run = async(files, diffForFile, warn) => {
```
Comment (raphaelboukara) [inline src/bindingPryPresenceFail/index.js]: fail

File: src/bindingPryPresenceFail/index.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    if (!files) {
+        return;
+    }
+
+    for (const file of files) {
+        const { after } = await diffForFile(file);
+
+        if (bindingPryDetected(after)) {
+            warn(MESSAGE);
```
Comment (raphaelboukara) [inline src/bindingPryPresenceFail/index.js]: fail

**Review by raphaelboukara — CHANGES_REQUESTED** [review]

**Review by raphaelboukara — APPROVED** [review]

**Pull request merged**